### PR TITLE
Feat/test api keys card component

### PR DESCRIPTION
**Pull request by yk-jp**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- This PR adds a component testing and a few code for an ApiKeysCard component.
- A few changes of code are added to make testing easier.

- **Why this change was needed?** (You can also link to an open issue here)
issue = #1344

- **Other information**:
To find a target element easily while testing, I added the label `data-test-id`.
If you find that something is missing or wrong, please inform me about it.

### CODE REVIEW

File: apps/web/src/pages/settings/tabs/ApiKeysCard.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const ApiKeysCard = () => {
             readOnly
             type={'password'}
             rightSection={
-              <Tooltip data-test-id={'Tooltip'} label={clipboardApiKey.copied ? 'Copied!' : 'Copy Key'}>
+              <Tooltip data-test-id={'api-key-Tooltip'} label={clipboardApiKey.copied ? 'Copied!' : 'Copy Key'}>
```
Comment (p-fernandez) [inline apps/web/src/pages/settings/tabs/ApiKeysCard.tsx:30]:
```suggestion
              <Tooltip data-test-id={'api-key-tooltip'} label={clipboardApiKey.copied ? 'Copied!' : 'Copy Key'}>
```
Seems we are mostly following a pattern of kebab-case data-test-id all in lowercase. Would you mind to update?

File: apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  updatedAt: '2022-10-03T04:43:47.460Z',
+  widget: {
+    notificationCenterEncryption: false,
+  },
+  __v: 0,
+  _id: '633a68831c6bf8ce582c923e',
+  _organizationId: '633a68831c6bf8ce582c9238',
+};
+
+const apiKeys: { key: string }[] = [
```
Comment (p-fernandez) [inline apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx:31]:
```suggestion
const apiKeys: IApiKey[] = [
```
This should be the right interface for the API keys. Also you would need to include an `userId`. You can double check it here: https://github.com/novuhq/novu/blob/bd7c2490725f031cb832ab8d801ad61ae9e293fc/libs/dal/src/repositories/environment/environment.entity.ts#L1

File: apps/web/src/pages/settings/tabs/ApiKeysCard.tsx (hunk 1 above)
Comment (yk-jp) [reply apps/web/src/pages/settings/tabs/ApiKeysCard.tsx]: Sorry, I was careless about it. 
It's fixed at 4a2068b1912b7c22ef50872864eb3d8cbc9d17be

File: apps/web/src/pages/settings/tabs/ApiKeysCard.tsx [hunk 3]
```diff
@@ -47,7 +47,10 @@ export const ApiKeysCard = () => {
           <Input
             readOnly
             rightSection={
-              <Tooltip label={clipboardEnvironmentIdentifier.copied ? 'Copied!' : 'Copy Key'}>
+              <Tooltip
+                data-test-id={'api-identifier-Tooltip'}
```
[the remark this answers is not in the document]
Comment (yk-jp) [reply apps/web/src/pages/settings/tabs/ApiKeysCard.tsx]: Sorry, I was careless about it. 
It's fixed at 4a2068b1912b7c22ef50872864eb3d8cbc9d17be

File: apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx (hunk 2 above)
Comment (yk-jp) [reply apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx]: I've updated apiKeys type at 713616c5cb18ba43df64bd7cafffff364fa2b6c9.

I've got a question about importing the interface.

I tried to import `IApiKey` by `` import { IApiKey } from '@novu/dal'``, but the import usage seems to be restricted by eslint. Is this what you expected?

**Review by yk-jp — COMMENTED** [review]
Should I apply the interface `IApiKey` here as well?

File: apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx (hunk 2 above)
Comment (p-fernandez) [reply apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx]: Just skip this request for this PR as we have realised it needs some more sensible changes. 👍🏻

File: apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx [hunk 4]
```diff
@@ -79,12 +79,17 @@ describe('Input Component', () => {
 
   it('should copy Api Key and Application Identifier to the clipboard', () => {
     cy.wait('@getApiKeys');
+    cy.window().then((win) => {
+      cy.spy(win.navigator.clipboard, 'writeText').as('copy');
+
+      return;
+    });
     cy.get("[data-test-id='api-key-tooltip']").click();
-    cy.window().its('navigator.clipboard').invoke('readText').should('equal', apiKeys[0].key);
```
Comment (yk-jp) [inline apps/web/src/pages/settings/tabs/ApiKeysCard.cy.tsx]: I changed the way to test the clipboard functionality.
I found that the clipboard is available only if permission is granted and `readText` seems to not work expectedly in cypress.

**Review by scopsy — APPROVED** [review]

**Pull request merged**